src/Proto.Cluster/ClusterExtension.cs

```diff
@@ -105,10 +106,10 @@ bool TryGetRef(MessageEnvelope envelope, out PidRef pidRef)
                     {
                         var pid = envelope.Sender;
 
-                        if (pid is not null && int.TryParse(pid.Id[1..], out var id) &&
+                        if (pid is not null && pid.RequestId > 0 && int.TryParse(pid.Id[1..], out var id) &&
```

Comment:
>int.TryParse(pid.Id[1..], out var id)

We parse the $123 id for the future and what do we do with the 123 part then?

Reply:
It becomes part of the PidRef, a cheaper way to store the Pid than as a string.

Reply:
And if it is not possible to parse, it is not an unnamed Future-pid, which is what we are checking for in the filter.